### PR TITLE
Automate publishing to PyPI and gh-pages

### PR DESCRIPTION
**Pull request by c-w**
This change automates the release to PyPI and gh-pages: whenever a new tag is pushed or a new release is authored in the Github UI, a new version of the project automatically gets published.

### CODE REVIEW

**Comment by ryan-roemer** [comment]
Nice automation work? How do we determine what semver / "next version" to release at? Or does a PR have to update the version string in `__init__.py`?

**Comment by c-w** [comment]
Good catch, fixed in faf6075.

**Comment by ryan-roemer** [comment]
I think I understand what's happening here now, but just to be sure, can you walk me through the lifecycle of things a human does from their computer and then after things Travis does to github and PyPI once we decide to cut a release from say current `master`?

Thanks!

**Comment by c-w** [comment]
Added the release publishing instructions in e3ee132. Once this PR is merged, all that a human will have to do to publish a new release is to [go here](https://github.com/ryan-roemer/django-cloud-browser/releases/new), name the release tag like `1.2.3`, optionally give the release a name and description and then click "Publish release". This will create a tag on the repo and kick off a Travis CD build.

The CD build then does:
1. Update `cloud_browser/__init__.py` with the version number from `TRAVIS_TAG`.
2. Build the source dist package.
3. Upload the dist package to PyPI.
4. Build the sphinx docs.
5. Push the docs folder to `gh-pages`.

**Comment by ryan-roemer** [comment]
Thanks! That all sounds good with just one remaining question: how does the code change to `cloud_browser/__init__.py` get committed back to `master` in git?

**Comment by c-w** [comment]
The code change currently doesn't get pushed back to master. The idea here is that it's not a real code change (it's really metadata for the PyPI release) so there's no need for the exact version information to be in the source tree (it would be redundant with the PyPI release history and the git tags). I've made this more explicit in `__init__.py` in 98545bf.

**Comment by ryan-roemer** [comment]
It's been a while since I've attempted this, but don't we want to support git-based dependencies on this project wherein the `cloud_browser/__init__.py` should have a correct version entry? (That's what we always do in npm/javascript land).

Usually in JS land we do something like:

```sh
$ npm version major|minor|patch|ACTUAL_VERSION_STRING
# updates `package.json:version`, commits to git and does `git tag -a `vVERSION`

$ npm publish
# dependency available from npm as `yarn add foo@v1.2.3`

$ git push && git push --tags
# dependency available from github as `yarn add ORG/foo#v1.2.3`
```

Can you walk me through modern expectations in the Python community for the git `master` version of an `__init__.py` file not being in sync with PyPI?

Thanks!

**Comment by c-w** [comment]
When doing a git-based install, one would usually specify something like `https://github.com/ryan-roemer/django-cloud-browser/archive/735ef1d.zip` in the `requirements.txt` file which will fetch the project files for that SHA, run `setup.py install`, etc. With the changes proposed in this pull request, the project would be set up as `django-cloud-browser==0.0.0` which should be fine since this is not a PyPI-published version making it explicit that this is not a normal install and the developer can still find the exact code they installed by looking at `735ef1d` on Github. `setup.py install` doesn't really care about what is in the version string as long as there's something there (e.g. I'm using this same approach in [another project](https://github.com/ascoderu/opwen-webapp/blob/70b4b5a3deaddea814c9c93d4e7c01d511fe80e7/opwen_email_client/__init__.py#L1) and it's been working fine).

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    export TWINE_PASSWORD="..."
+
+    # set a personal access token with push access to the repo
+    export GITHUB_TOKEN="..."
+
+    # set the new version number
+    export BUILD_VERSION="1.2.3"
+
+    # run the release
+    fab sdist publish_pypi docs publish_docs
```
Comment (ryan-roemer) [inline CONTRIBUTING.rst]: Does this modify the in-source file? If so, can we also add a `git checkout cloud_browser/__init__.py` or something revert step manually here (or in the fab wrapper)?

Comment (c-w) [reply CONTRIBUTING.rst]: Great idea. Done in 467db66.

**Comment by c-w** [comment]
@ryan-roemer Any suggestions for what the next release's version number should be? `0.5.0`?

**Comment by ryan-roemer** [comment]
`0.5.0` sounds good.

**Pull request merged**